crates/gen/src/numerics.rs

```diff
@@ -0,0 +1,190 @@
+use crate::format_ident;
+use crate::types::*;
+use proc_macro2::TokenStream;
+use quote::quote;
+use std::iter::FromIterator;
+
+pub fn generate_struct_extensions(winrt_struct: &Struct) -> TokenStream {
```

Comment:
So this seems nice and generic in terms of how it iterates the struct fields and so on, but I'm wondering whether this is overkill since the numerics can never change and are unlikely to ever grow. Perhaps we can simply write out the numeric operations by hand and simply include them all in a single quote! (per type). I suspect that will be easier to reason about.

Reply:
Fair enough, I'll run this and paste the output in directly. I just didn't want to type all of it by hand 🙂